lib/appcanary/configuration.rb

```diff
@@ -0,0 +1,51 @@
+require "yaml"
+
+module Appcanary
+  APPCANARY_DEFAULT_BASE_URI = "https://appcanary.com/api/v3/monitors"
+
+  class Configuration
+    attr_accessor :base_uri, :api_token, :monitor_name
+
+    def initialize
+      self.base_uri = APPCANARY_DEFAULT_BASE_URI
+    end
+  end
+
+  class ConfigurationError < RuntimeError
+  end
+
+  class << self
```

Comment:
Right - that's my understanding. My concern is that I turned a module into a bunch of callable static methods by putting them all into the eigenclass for the module. I'm not sure if that's ok, or if maybe it's more idiomatic to make a special class to use programmatically and extend it with all the module methods.

Something like

```ruby
module Appcanary
  class Client
    extend Appcanary
  end
end
```

and maybe sidestep the `class << self` altogether, so one can instead ask

```ruby
Appcanary::Client.vulnerable?
```

Maybe I try it.